src/ucs/datastruct/strided_alloc.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2018.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_STRIDED_ALLOC_H_
+#define UCS_STRIDED_ALLOC_H_
+
+#include "queue_types.h"
+
+#include <ucs/type/status.h>
+#include <ucs/sys/compiler_def.h>
+#include <ucs/sys/math.h>
+#include <stddef.h>
+
+
+BEGIN_C_DECLS
+
+
+/* the distance between allocated elements */
+#define UCS_STRIDED_ALLOC_STRIDE (128 * UCS_KBYTE)
+
+
+/**
+ * Get a pointer to another element in the strided object
+ *
+ * Example with stride_count=3:
+ *
+ * chunk
+ * start -+
+ *        |
+ *        \/
+ *        |
+ *        +--------+--   ...   --+--------+--   ...   --+--------+
+ *        | stride |             | stride |             | stride |
+ * obj0:  | elem 0 | <- 128kB -> | elem 1 | <- 128kB -> | elem 2 |
```

Comment:
`128kB - sizeof(elem <N>)`, right?

Reply:
it's UCS_STRIDED_ALLOC_STRIDE

Reply:
As I understand UCS_STRIDED_ALLOC_STRIDE is a distance between base of `elem 0` (not its end) and `elem 1`, so it includes size of elem <N>

Reply:
right, the drawing is incorrect, will fix